src/core/jit/synth/synth.rs

```diff
@@ -128,10 +142,22 @@ where
         }
 
         let eval_result = if value.is_null() {
-            if node.type_of.is_nullable() {
-                Ok(Value::null())
-            } else {
-                Err(ValidationError::ValueRequired.into())
+            // link to GraphQL reference https://spec.graphql.org/October2021/#sec-Handling-Field-Errors
+            match &node.type_of {
+                crate::core::blueprint::Type::NamedType { name: _, non_null: _ } => {
```

Comment:
do we need a match here or we can use `is_nullable` on the type_of itself?

Reply:
We want to test the inside of the type in case it is a list.
If we have `[Foo!]` in this scenario, I check if Foo is nullable or not, so I need to unwrap